Electron 9.0.0-beta.16 crashes when importing node-keytar

<!--

Have you read Atom's Code of Conduct? By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/atom/.github/blob/master/CODE_OF_CONDUCT.md

Do you want to ask a question? Are you looking for support? The Atom message board is the best place for getting support: https://discuss.atom.io

-->

### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description
Electron 9.0.0-beta.16 crashes with exit code `3221225477` when just importing `node-keytar` rebuilt using electron-builder.

### Steps to Reproduce

Just type the following in an Electron app.
```ts
require('keytar');
```

**Reproduces how often:**

Always

### Versions

* Windows 10
* node-keytar 5.4.0
* Electron 9.0.0-beta.16

To get a runnable example I made these changes to `electron-quick-start`, importing the module as late as possible:

```
$ git rev-parse HEAD
d2db0d4cc1f7ebcf6c562eeb5d5e3a8eabd0432e

$ git diff -- package.json main.js
diff --git a/main.js b/main.js
index c4d7e19..bce4a7e 100644
--- a/main.js
+++ b/main.js
@@ -3,6 +3,10 @@ const {app, BrowserWindow} = require('electron')
 const path = require('path')

 function createWindow () {
+  console.log(`Loading keytar...`);
+  const keytar = require('keytar');
+  console.log(`Loaded keytar, displaying window...`);
+
   // Create the browser window.
   const mainWindow = new BrowserWindow({
     width: 800,
diff --git a/package.json b/package.json
index f747f34..a4c5bf5 100644
--- a/package.json
+++ b/package.json
@@ -16,7 +16,10 @@
   ],
   "author": "GitHub",
   "license": "CC0-1.0",
+  "dependencies": {
+    "keytar": "^5.5.0"
+  },
   "devDependencies": {
-    "electron": "^8.2.3"
+    "electron": "9.0.0-beta.19"
   }
 }
```

I then `npm i && npm start` to launch the app and it does get stuck, but in a different way:

```
$ npm i && npm start

> keytar@5.5.0 install C:\Users\shiftkey\src\electron-quick-start\node_modules\keytar
> prebuild-install || node-gyp rebuild


> core-js@3.6.5 postinstall C:\Users\shiftkey\src\electron-quick-start\node_modules\core-js
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon:
> https://opencollective.com/core-js
> https://www.patreon.com/zloirock

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> electron@9.0.0-beta.19 postinstall C:\Users\shiftkey\src\electron-quick-start\node_modules\electron
> node install.js

added 134 packages from 132 contributors and audited 202 packages in 6.038s
found 0 vulnerabilities


> electron-quick-start@1.0.0 start C:\Users\shiftkey\src\electron-quick-start
> electron .


Loading keytar...
(node:7060) UnhandledPromiseRejectionWarning: Error: Module did not self-register: '\\?\C:\Users\shiftkey\src\electron-quick-start\node_modules\keytar\build\Release\keytar.node'.
    at process.func (electron/js2c/asar.js:140:31)
    at process.func [as dlopen] (electron/js2c/asar.js:140:31)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1034:18)
    at Object.func (electron/js2c/asar.js:140:31)
    at Object.func [as .node] (electron/js2c/asar.js:140:31)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Module._load (electron/js2c/asar.js:738:28)
    at Function.Module._load (electron/js2c/asar.js:738:28)
    at Module.require (internal/modules/cjs/loader.js:852:19)
(node:7060) UnhandledPromiseRejectionWarning: Error: Module did not self-register: '\\?\C:\Users\shiftkey\src\electron-quick-start\node_modules\keytar\build\Release\keytar.node'.
    at process.func (electron/js2c/asar.js:140:31)
    at process.func [as dlopen] (electron/js2c/asar.js:140:31)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1034:18)
    at Object.func (electron/js2c/asar.js:140:31)
    at Object.func [as .node] (electron/js2c/asar.js:140:31)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Module._load (electron/js2c/asar.js:738:28)
    at Function.Module._load (electron/js2c/asar.js:738:28)
    at Module.require (internal/modules/cjs/loader.js:852:19)
(node:7060) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:7060) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:7060) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:7060) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This is because `npm` is going to use the installed version of `node` to obtain any prebuilt binaries, so I added an `.npmrc` file to ensure we're using the right version when `prebuild-install` runs:

```
$ git add .npmrc

$ cat .npmrc
runtime = electron
disturl = https://atom.io/download/electron
target = 9.0.0-beta.19
arch = x64

$ git clean -xdf
Removing node_modules/

$ npm i

> keytar@5.5.0 install C:\Users\shiftkey\src\electron-quick-start\node_modules\keytar
> prebuild-install || node-gyp rebuild


> core-js@3.6.5 postinstall C:\Users\shiftkey\src\electron-quick-start\node_modules\core-js
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon:
> https://opencollective.com/core-js
> https://www.patreon.com/zloirock

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> electron@9.0.0-beta.19 postinstall C:\Users\shiftkey\src\electron-quick-start\node_modules\electron
> node install.js

added 134 packages from 132 contributors and audited 202 packages in 14.837s
found 0 vulnerabilities
```

Cool, still looks to be using the prebuild version. Let's run it again:

```
$ npm start

> electron-quick-start@1.0.0 start C:\Users\shiftkey\src\electron-quick-start
> electron .


Loading keytar...
npm ERR! code ELIFECYCLE
npm ERR! errno 3221225477
npm ERR! electron-quick-start@1.0.0 start: `electron .`
npm ERR! Exit status 3221225477
npm ERR!
npm ERR! Failed at the electron-quick-start@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\shiftkey\AppData\Roaming\npm-cache\_logs\2020-04-26T13_37_47_447Z-debug.log
/c/Program Files/nodejs/npm: line 37:   684 Segmentation fault      "$NODE_EXE" "$NPM_CLI_JS" "$@"
```

Running it outside of `npm` might help with yielding how the native module is failing, but for now that helps me get up to speed.

--- a/package.json
+++ b/package.json
@@ -36,8 +36,8 @@
     "test": "npm run lint && npm build . && mocha --require babel-core/register spec/",
     "prebuild-node": "prebuild -t 8.9.0 -t 9.4.0 -t 10.11.0 -t 11.9.0 -t 12.0.0 -t 13.0.0 --strip",
     "prebuild-node-ia32": "prebuild -t 8.9.0 -t 9.4.0 -a ia32 --strip",
-    "prebuild-electron": "prebuild -t 5.0.0 -t 6.0.0 -t 7.0.0 -t 8.0.0 -t 9.0.0-beta.2 -r electron --strip",
-    "prebuild-electron-ia32": "prebuild -t 5.0.0 -t 6.0.0 -t 7.0.0 -t 8.0.0 -t 9.0.0-beta.2 -r electron -a ia32 --strip",
+    "prebuild-electron": "prebuild -t 5.0.0 -t 6.0.0 -t 7.0.0 -t 8.0.0 -t 9.0.0-beta.16 -r electron --strip",
+    "prebuild-electron-ia32": "prebuild -t 5.0.0 -t 6.0.0 -t 7.0.0 -t 8.0.0 -t 9.0.0-beta.16 -r electron -a ia32 --strip",
     "upload": "node ./script/upload.js",
     "postpublish": "git push --follow-tags"
   },

